src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1005,26 +1008,32 @@ public TextLoader(IHostEnvironment env, Arguments args, IMultiStreamSource files
                 _inputSize = SrcLim - 1;
 
             _host.CheckNonEmpty(args.Separator, nameof(args.Separator), "Must specify a separator");
-            string sep = args.Separator.ToLowerInvariant();
+            _host.CheckNonEmpty(args.Delimiter.ToString(), nameof(args.Delimiter), "Must specify a delimeter");
 
-            if (sep == ",")
-                _separators = new char[] { ',' };
+            if (args.Delimiter != '\t')
+                _separators = new char[] { args.Delimiter };
```

Comment:
Maybe comment on the code how this relies on the fact that the default `arg.Separator` is `"tab"`, since to a reader this will look odd.